### PR TITLE
fix rotl.i16 with i128 shift value.

### PR DESCRIPTION
**Pull request by yuyang-ok**
This fix issue #5523.

### CODE REVIEW

**Comment by hch12907** [comment]
The title and commit messages could be changed to "fix rotl.i16 with i128 shift value" or something similar.

**Comment by yuyang-ok** [comment]
@hch12907 Ok,I Think the works the same.

**Comment by yuyang-ok** [comment]
@afonso360 are you going to review this??

**Review by afonso360 — COMMENTED** [review]
Hey, Someone else has to review this in order to get it merged. But I'll run the fuzzer on this and report back if it finds anything else.

File: cranelift/codegen/src/isa/riscv64/inst.isle
Comment (yuyang-ok) [reply cranelift/codegen/src/isa/riscv64/inst.isle:1359]: ok, thanks

File: cranelift/codegen/src/isa/riscv64/inst.isle
Comment (cfallin) [inline cranelift/codegen/src/isa/riscv64/inst.isle:1251]: This comment is outdated now, I think? This implementation does seem to support shifting by >= 64 bits, with the selects below.

File: cranelift/filetests/filetests/runtests/issue5523.clif
Comment (cfallin) [inline cranelift/filetests/filetests/runtests/issue5523.clif:3]: Just for good coverage while we're here, can we add the other targets (`x86_64`, `aarch64`, `s390x`) here too?

**Review by cfallin — APPROVED** [review]
Thanks! One out-of-date comment to remove and a request to apply the runtest to other ISAs too, otherwise LGTM.

File: cranelift/filetests/filetests/runtests/issue5523.clif
Comment (yuyang-ok) [reply cranelift/filetests/filetests/runtests/issue5523.clif:3]: Ok

**Comment by cfallin** [comment]
Looks like a [test failure](https://github.com/bytecodealliance/wasmtime/actions/runs/4049023616/jobs/6964942709#step:15:789) in an expected-output test. Once that is fixed, I'm happy to merge!

**Comment by yuyang-ok** [comment]
@cfallin Yes,I am trying to fix this.

**Comment by yuyang-ok** [comment]
image
This failure related to x86_64 backend???

**Comment by cfallin** [comment]
Ah, yes, `i128` in runtests on x86-64 requires `set enable_llvm_abi_extensions=true` before the `target` lines. You can see examples in other runtests (e.g., `i128-call.clif`). I think once you add that, it should pass.

**Comment by yuyang-ok** [comment]
@cfallin Thanks.

**Comment by cfallin** [comment]
I filed #5680 for the failing test here, as that's an issue on `main` too. Once we have a fix for that, we can rebase and then merge this I think.

**Comment by yuyang-ok** [comment]
@cfallin ok.

**Comment by cfallin** [comment]
I just pushed a merge commit onto this branch with the latest `main` including fixes; should pass now, will merge if so.

**Comment by yuyang-ok** [comment]
@cfallin cool.

**Pull request merged**
